### PR TITLE
Short next beacon time on beacon error

### PR DESCRIPTION
**Pull request by madninja**
This constructs a short next beacon time when there is a failure to construct, transmit or (finally) deliver a beacon report.

### CODE REVIEW

File: src/beaconer.rs [hunk 1]
```diff
@@ -157,32 +156,29 @@ impl Beaconer {
...
-            Ok(report) => report,
-            Err(err) => {
-                warn!(beacon_id, %err, "poc beacon report");
-                return;
-            }
-        };
-        let _ = PocIotService::new(self.poc_ingest_uri.clone())
+        PocIotService::new(self.poc_ingest_uri.clone())
             .submit_beacon(report)
             .inspect_err(|err| info!(beacon_id, %err, "submit poc beacon report",))
```
Comment (mawdegroot) [inline src/beaconer.rs:177]: Is the drop from warn! to info! intentional?

File: src/beaconer.rs [hunk 2]
```diff
@@ -213,21 +209,21 @@ impl Beaconer {
...
-        };
+        let interval = self.interval;
+        let (last_beacon, next_beacon_time) = self
+            .mk_beacon()
+            .inspect_err(|err| warn!(%err, "construct beacon"))
+            .and_then(|beacon| self.send_beacon(beacon))
+            // On success to construct and transmit a beacon and it's report
+            // select a normal full next beacon time
+            .map_ok(|beacon| (Some(beacon), Self::mk_next_beacon_time(interval)))
+            // On failure to construct, transmit or send a beacon or it's
```
Comment (ke6jjj) [inline src/beaconer.rs:220]: While we're here .... `s/it's/its/`

File: src/beaconer.rs [hunk 3]
```diff
@@ -265,47 +261,6 @@ impl Beaconer {
...
             })
             .await;
-
-        // Disable secondary beacons until TTL is implemented
-        if false {
-            self.handle_secondary_beacon(report).await
-        }
-    }
-
-    async fn handle_secondary_beacon(&mut self, report: poc_lora::LoraWitnessReportReqV1) {
```
Comment (ke6jjj) [inline src/beaconer.rs:275]: Ah secondary beacons, how we will miss even the code that showed you once existed.

Comment (madninja) [reply src/beaconer.rs:275]: lol.. got too annoying to maintain unused code

File: src/beaconer.rs (hunk 2 above)
Comment (madninja) [reply src/beaconer.rs]: fixed in [c13647b](https://github.com/helium/gateway-rs/pull/434/commits/c13647b1071e8cf1475b52f66b792846e89f2814)

**Pull request merged**